### PR TITLE
API Alignment

### PR DESCRIPTION
**Pull request by DariusIMP**
This PR extensively updates the Zenoh Java API.

### Removing all builder patterns and resolvables

We noticed this kind of API was causing confusions among Java users:
```java
publisher.put(payload).attachment(attachment).res();
```
For 2 reasons:
- the imperativeness of statements such as "put" doesn't make you think it's in fact returning a builder
- the `.res()` call was often forgotten 

It was decided to go for an approach similar to the C++'s bindings with Option objects, for instance:

```java
PublisherOptions options = new PublisherOptions();
options.setEncoding(Encoding.ZENOH_STRING);
options.setReliability(Reliability.RELIABLE);
Publisher publisher = session.declarePublisher(keyExpr, config);
```

## New features introduced:
- Scouting
- Liveliness
- ZBytes (replacing Value)
- SessionInfo

## Documentation:

All the documentation was updated with examples following the new API.

### Logger refactoring

Logs can now be achieved by running for instance:
```
RUST_LOG=debug gradle ZGet
```
similar to other Zenoh bindings.

### Packages reordering

- Aligning the sorting of classes and packages with the other bindings

### Adding new examples
 
- Added examples: Zping, ZPong, ZScout

- Also providing argument handling to them with 'picocli'.

## Internal:

- Porting tests from kotlin to java
- Adoption of the "@JvmOverload" tag 

## Missing items

ZBytes serialization and deserialization utilities, to be added in a followup PR.

### CODE REVIEW

File: zenoh-java/src/commonMain/kotlin/io/zenoh/query/Get.kt [hunk 1]
```diff
@@ -14,181 +14,25 @@
...
-                    target,
-                    consolidation,
-                    value,
-                    attachment
-                )
-            }
-        }
-    }
-}
+data class GetOptions(
```
Comment (milyin) [inline zenoh-java/src/commonMain/kotlin/io/zenoh/query/Get.kt:31]: There are no getters/setters in GetOptions. Corresponding example also doesn't demonstrate `GetOptions` usage

File: zenoh-java/src/commonMain/kotlin/io/zenoh/pubsub/DeleteOptions.kt [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+import io.zenoh.qos.Reliability
+
+/**
+ * Options for delete operations.
+ *
+ * @param qos The [QoS] (Quality of Service) desired.
+ * @param reliability The [Reliability] desired.
+ * @param attachment Optional attachment for the delete operation.
+ */
+data class DeleteOptions(
```
Comment (milyin) [inline zenoh-java/src/commonMain/kotlin/io/zenoh/pubsub/DeleteOptions.kt:28]: No getters/setters, unlike e.g. `PutOptions`

**Review by milyin — CHANGES_REQUESTED** [review]
Looks good, but some options structures doesn't implement setters/getters.

File: zenoh-java/src/commonMain/kotlin/io/zenoh/query/Get.kt (hunk 1 above)
Comment (DariusIMP) [reply zenoh-java/src/commonMain/kotlin/io/zenoh/query/Get.kt:31]: Ok I'll add the demonstration to the Get example, but we don't need getters and setters for data classes, Kotlin implements them for you. The reason you see some getters and setters in other data classes is for the QoS parameters, just as a utility.

**Comment by DariusIMP** [comment]
I just changed a couple of things:
- Added the GetOptions instance on the ZGet example
- Refactored the Options classes regarding the QoS config

**Review by milyin — APPROVED** [review]

**Pull request merged**